test/validators/remove.ts

```diff
@@ -23,205 +23,416 @@ describe('Remove Validator Tests', () => {
+  it('Remove validator with an invalid operator setup reverts "ClusterDoesNotExists"', async () => {
+    await expect(
+      ssvNetworkContract
+        .connect(helpers.DB.owners[1])
+        .removeValidator(helpers.DataGenerator.publicKey(1), [1, 2, 3, 5], firstCluster.cluster),
+    ).to.be.revertedWithCustomError(ssvNetworkContract, 'ClusterDoesNotExists');
   });
 
   it('Remove the same validator twice reverts "ValidatorDoesNotExist"', async () => {
     // Remove validator
-    await trackGas(ssvNetworkContract.connect(helpers.DB.owners[1]).removeValidator(
-      helpers.DataGenerator.publicKey(1),
-      firstCluster.operatorIds,
-      firstCluster.cluster
-    ), [GasGroup.REMOVE_VALIDATOR]);
+    const result = await trackGas(
+      ssvNetworkContract
+        .connect(helpers.DB.owners[1])
+        .removeValidator(helpers.DataGenerator.publicKey(1), firstCluster.operatorIds, firstCluster.cluster),
+      [GasGroup.REMOVE_VALIDATOR],
+    );
+
+    const removed = result.eventsByName.ValidatorRemoved[0].args;
+
+    // Remove validator again
+    await expect(
+      ssvNetworkContract
+        .connect(helpers.DB.owners[1])
+        .removeValidator(helpers.DataGenerator.publicKey(1), removed.operatorIds, removed.cluster),
+    ).to.be.revertedWithCustomError(ssvNetworkContract, 'ValidatorDoesNotExist');
+  });
+
+  it('Remove the same validator with wrong input parameters reverts "IncorrectClusterState"', async () => {
+    // Remove validator
+    await trackGas(
+      ssvNetworkContract
+        .connect(helpers.DB.owners[1])
+        .removeValidator(helpers.DataGenerator.publicKey(1), firstCluster.operatorIds, firstCluster.cluster),
+      [GasGroup.REMOVE_VALIDATOR],
+    );
 
     // Remove validator again
-    await expect(ssvNetworkContract.connect(helpers.DB.owners[1]).removeValidator(
-      helpers.DataGenerator.publicKey(1),
-      firstCluster.operatorIds,
-      firstCluster.cluster
-    )).to.be.revertedWithCustomError(ssvNetworkContract, 'ValidatorDoesNotExist');
+    await expect(
+      ssvNetworkContract
+        .connect(helpers.DB.owners[1])
+        .removeValidator(helpers.DataGenerator.publicKey(1), firstCluster.operatorIds, firstCluster.cluster),
+    ).to.be.revertedWithCustomError(ssvNetworkContract, 'IncorrectClusterState');
+  });
+
+  it('Bulk Remove validator that does not exist in a valid cluster reverts "ValidatorDoesNotExist"', async () => {
+    const { args, pks } = await helpers.bulkRegisterValidators(
+      2,
+      10,
+      helpers.DEFAULT_OPERATOR_IDS[4],
+      minDepositAmount,
+      helpers.getClusterForValidator(0, 0, 0, 0, true),
+    );
+
+    pks[2] = "0xabcd1234";
+
+    await expect(
+      ssvNetworkContract
+        .connect(helpers.DB.owners[2])
+        .bulkRemoveValidator(pks, args.operatorIds, args.cluster),
+    ).to.be.revertedWithCustomError(ssvNetworkContract, 'ValidatorDoesNotExist');
+  });
```

Comment:
From your updates in the `bulkRemoveValidators()` function in the **SSVClusters.sol**, changed error code.
**AssertionError**: Expected transaction to be reverted with custom error **`ValidatorDoesNotExist`**, but it reverted with custom error **`IncorrectValidatorStateWithData`**